Tests/Algorithm/AlgorithmAddDataTests.cs

```diff
@@ -171,6 +174,310 @@ public void OnEndOfTimeStepDoesNotThrowWhenSeedsSameUnderlyingForTwoSecurities()
             Assert.AreEqual(data.Price, 2);
         }
 
+        [TestCase("EURUSD", typeof(PsychSignalSentimentData), SecurityType.Cfd, false, true)]
+        [TestCase("BTCUSD", typeof(PsychSignalSentimentData), SecurityType.Crypto, false, true)]
+        [TestCase("CL", typeof(PsychSignalSentimentData), SecurityType.Future, false, true)]
+        [TestCase("EURUSD", typeof(PsychSignalSentimentData), SecurityType.Forex, false, true)]
+        [TestCase("AAPL", typeof(PsychSignalSentimentData), SecurityType.Equity, true, true)]
+        [TestCase("EURUSD", typeof(TradingEconomicsCalendar), SecurityType.Cfd, false, false)]
+        [TestCase("BTCUSD", typeof(TradingEconomicsCalendar), SecurityType.Crypto, false, false)]
+        [TestCase("CL", typeof(TradingEconomicsCalendar), SecurityType.Future, false, false)]
+        [TestCase("AAPL", typeof(TradingEconomicsCalendar), SecurityType.Equity, true, false)]
+        [TestCase("EURUSD", typeof(TradingEconomicsCalendar), SecurityType.Forex, false, false)]
+        public void AddDataSecuritySymbolWithUnderlying(string ticker, Type customDataType, SecurityType securityType, bool securityShouldBeMapped, bool customDataShouldBeMapped)
```

Comment:
For these new unit tests maybe `customDataShouldBeMapped` parameter could be retrieved directly from the custom data type since that's why Lean is going to be doing using `dataType.GetBaseDataInstance().RequiresMapping()`